File: 卡码/代码随想录/1/06-二叉树/Go/05-二叉树公共祖先/LeetCode_0235.go
//235. 二叉搜索树的最近公共祖先
package main

import (
	. "binaryTree/types"
)

func lowestCommonAncestor(root, p, q *TreeNode) *TreeNode {
	for root!=nil {
		if root.Val>p.Val&&root.Val>q.Val {
			root=root.Left
		}else if root.Val<p.Val&&root.Val<q.Val {
			root=root.Right
		}else{
			return root
		}
	}
	return nil
}

File: 卡码/代码随想录/1/06-二叉树/Js/05-二叉树公共祖先/LeetCode_0235.js
//235. 二叉搜索树的最近公共祖先
import TreeNode from "../TreeNode.js";

const lowestCommonAncestor = function(root, p, q) {
  while(root){
    if(root.val>p.val&&root.val>q.val) root=root.left;
    else if(root.val<p.val&&root.val<q.val) root=root.right;
    else return root;
  }
  return null;
};